tests/Feature/Http/Controllers/PostControllerTest.php

```diff
@@ -2,27 +2,167 @@
 
 namespace Tests\Feature\Http\Controllers;
 
-use App\Models\Category;
+use Tests\TestCase;
+use App\Models\Post;
 use App\Models\User;
-use Illuminate\Foundation\Testing\RefreshDatabase;
+use App\Models\Category;
 use Illuminate\Foundation\Testing\WithFaker;
-use Tests\TestCase;
+use Illuminate\Foundation\Testing\RefreshDatabase;
 
 class PostControllerTest extends TestCase
 {
     use RefreshDatabase;
     use WithFaker;
 
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->user = User::factory()->create()->assignRole('user');
```

Comment:
You forgot to declare the properties user, moderator, category and post: `private $user;` ...